HashSet.py

```diff
@@ -0,0 +1,74 @@
+"""
+// Time Complexity : O(1) for all methods
+// Space Complexity : O(n + log(n) )
```

Comment:
why log(n)?

Reply:
Because sub-array is not created for every element in the primary array.